packages/create-emotion-styled/types/test.tsx

```diff
@@ -0,0 +1,92 @@
+import createEmotion from 'create-emotion';
+import React from 'react';
+
+import createEmotionStyled, { StyledComponent } from '../';
+
+const context = {
+  name: 'emotion',
+};
+
+const createStyled = createEmotionStyled(createEmotion(context), React);
+
+interface TestClassProps {
+  readonly some: number;
+  partial?: boolean;
+}
+
+class TestClassComp extends React.Component<TestClassProps> {
+  render() {
+    return (
+      <div>
+        {this.props.some}
+      </div>
+    );
+  }
+}
+
+interface TestFunProps0 {
+  complex: {
+    readonly id: number;
+    name: string;
+    friends: Array<number>;
+  };
+}
+
+const TestFunComp0 = (props: TestFunProps0) => (
+  <div>
+    {props.complex.friends.map((fid) => <div>{fid}</div>)}
+  </div>
+);
+
+interface TestFunProps1 extends TestFunProps0 {
+  world: number;
+}
+
+const TestFunComp1 = (props: TestFunProps1) => (
+  <div>
+    {props.world}
+    <TestFunComp0 complex={props.complex} />
+  </div>
+);
+
+const StyledClassComp0 = createStyled(TestClassComp)({
+  width: '200px',
+}, (props) => ({
+  height: props.theme ? '200px' : '100px',
```

Comment:
`props.theme` will never be falsy, if there isn't a `ThemeProvider`, `props.theme` will be an empty object.

Reply:
Oh, I didn't know that. I should fix typing also.